lib/handlers/subscribe.js

```diff
@@ -119,19 +136,31 @@ function addSubs (sub, done) {
   const broker = client.broker
   const topic = sub.topic
   const qos = sub.qos
+  const rh = this.rh
+  const rap = this.rap
+  const nl = this.nl
   let func = qos > 0 ? client.deliverQoS : client.deliver0
 
+  if (!rap) {
```

Comment:
@phil-mitchell can rap be undefined too or is it true by default? Would a check `rap === false` be better?

Reply:
@robertsLando rap  can be undefined, in which case it should be considered false.  i.e. rap = false is default.  So any falsey value of rap should set `retain = false`